sphinxcontrib/needs/directives/needpie.py

```diff
@@ -155,15 +155,23 @@ def process_needpie(app, doctree, fromdocname):
                     filter_func(**context)
                 sizes = context["results"]
                 # check items in sizes
+                if not isinstance(sizes, list):
+                    logger.error(
+                        f"The returned values {sizes} from the given filter_func {filter_func} is not valid."
```

Comment:
You should not print the content of "sizes" here.
You don't know how big it is and it may make the log message unreadable.

Reply:
ah you are right, will fix.